ci/create_yum_repo/main.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+import logging
+import os
+import sys
+from concurrent import futures
+
+import boto3
+import requests
+from jinja2 import Template
+from requests.adapters import HTTPAdapter
+
+import utils
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+
+def upload_directory(
+    directory: str,
+    prefix: str,
+    bucket: str = os.getenv("AWS_BUCKET_REPOS"),
+    boto3_session: boto3.Session = boto3.Session(),
+) -> int:
+    ret = 0
+    s3 = boto3_session.client("s3")
+
+    def error(e: Exception) -> None:
+        raise e
+
+    def walk_directory(directory: str) -> str:
+        for root, _, files in os.walk(directory, onerror=error):
+            for f in files:
+                yield os.path.join(root, f)
+
+    def upload_file(filename: str):
+        s3.upload_file(
+            Filename=filename,
+            Bucket=bucket,
+            Key=prefix + "/" + os.path.relpath(filename, directory),
+        )
+
+    with futures.ThreadPoolExecutor() as executor:
+        upload_task = {}
+
+        for filename in walk_directory(directory):
+            upload_task[executor.submit(upload_file, filename)] = filename
+
+        for task in futures.as_completed(upload_task):
+            try:
+                task.result()
+            except Exception as e:
+                logging.error(
+                    f"Exception {e} encountered while uploading file"
+                    f"{upload_task[task]}"
+                )
+                ret = 1
+    return ret
+
+
+def create_repo(repodir: str) -> int:
+    return utils.run_cmd(cmd=f"createrepo_c {repodir}")
+
+
+def test_repo(bucket: str, region: str, id: int) -> int:
+    with open("/etc/yum.repos.d/automotive.repo", "w") as repofile:
+        repofile.write(
+            Template(
+                open(
+                    os.path.join(
+                        (os.path.dirname(os.path.realpath(__file__))),
+                        "yumrepo.j2",
+                    ),
+                ).read()
+            ).render(bucket=bucket, region=region, id=id)
+        )
+    return utils.run_cmd(cmd="sudo dnf makecache --repo=automotive")
```

Comment:
Just a little nitpick. While I think using a string shouldn't be a problem in this case, a list would be cleaner.